frontend/packages/kubevirt-plugin/src/components/vm-status/use-vm-status-resources.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { PersistentVolumeClaimModel, PodModel } from '@console/internal/models';
+import {
+  K8sKind,
+  K8sResourceCommon,
+  PersistentVolumeClaimKind,
+  PodKind,
+} from '@console/internal/module/k8s';
+import { useDebounceCallback } from '../../hooks/use-debounce';
+import { useDeepCompareMemoize } from '../../hooks/use-deep-compare-memoize';
+import { DataVolumeModel, VirtualMachineInstanceMigrationModel } from '../../models';
+import { kubevirtReferenceForModel } from '../../models/kubevirtReferenceForModel';
+import { V1alpha1DataVolume } from '../../types/api';
+
+export type VmStatusResourcesValue = {
+  loaded: boolean;
+  loadError: string;
+  pods: PodKind[];
+  migrations: K8sKind[];
+  dvs: V1alpha1DataVolume[];
+  pvcs: PersistentVolumeClaimKind[];
+};
+
+export const useVmStatusResources = (namespace: string): VmStatusResourcesValue => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+  const [pods, setPods] = React.useState<PodKind[]>([]);
+  const [migrations, setMigrations] = React.useState<K8sKind[]>([]);
+  const [dvs, setDvs] = React.useState<V1alpha1DataVolume[]>([]);
+  const [pvcs, setPvcs] = React.useState<PersistentVolumeClaimKind[]>([]);
+
+  const watchedResources = React.useMemo(
```

Comment:
you do not have to use useMemo anymore, watchK8s hook is comparing the input args using deep compare